pkg/container-hook/privatedata.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package containerhook
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"syscall"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/link"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"golang.org/x/sys/unix"
+)
+
+//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -target $TARGET -cc clang -no-global-types privatedata ./bpf/privatedata.bpf.c -- -I./bpf/ -I../${TARGET}
+
+// ReadPrivateDataFromFd use ebpf to read the private_data pointer from the
+// kernel "struct file" associated with the given fd.
+//
+// It can then be used in other ebpf programs.
+//
+// This is done:
+//   - without using bpf iterators in order to work on old kernels.
+//   - without comparing pids from userspace and ebpf in order to work from
+//     different pid namespaces.
+func ReadPrivateDataFromFd(fd int) (uint64, error) {
```

Comment:
Should it call `runtime.LockOSThread` to avoid the (remote) possibility of another goroutine using `unix.Sendmsg (...UnixRights()))` at the same time and messing up with the data?

Reply:
> Do we need / want to export this function?

No need to export it now, I'll rename to `readPrivateDataFromFd`.